modules/admission-webhook-create.adoc

```diff
@@ -0,0 +1,328 @@
+    tls.key: /etc/pki/CA/serving-server.<namespace_name>.svc.key
+----
+<1> References the signed webhook server key and certificate.
+
+. Declare a service account:
++
+[source,yaml]
+----
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    namespace: <namespace_name>
+    name: server
+----
+
+. Expose the webhook server within the cluster:
++
+[source,yaml]
+----
+- apiVersion: v1
+  kind: Service
+  metadata:
+    namespace: <namespace_name>
+    name: server
+    annotations:
+      service.alpha.openshift.io/serving-cert-secret-name: server-serving-cert
+  spec:
+    selector:
+      server: "true"
+    ports:
+    - port: <host_port>
+      targetPort: <container_port>
+----
+
+. Define a custom resource definition for the webhook server:
++
+[source,yaml]
+----
+- apiVersion: apiextensions.k8s.io/v1beta1
+  kind: CustomResourceDefinition
+  metadata:
+    name: <resource_name>.online.openshift.io  <1>
+  spec:
+    group: online.openshift.io  <2>
+    version: v1alpha1  <3>
+    scope: Cluster  <4>
+    names:
+      plural: <resource_name>  <5>
+      singular: <resource_name_singular>  <6>
+      kind: <camel_case_resource_name>  <7>
+----
+<1> `<plural>.<group>` format and reflects custom resource definition `spec` values.
+<2> REST API group name.
+<3> REST API version name.
+<4> Accepted values are `Namespaced` or `Cluster`.
+<5> Plural name to be included in URL.
+<6> Alias seen in `oc` output.
+<7> Camel case reference for resource manifests.
+
+. Configure the admission webhook server also as an aggregated API server:
++
+[source,yaml]
+----
+- apiVersion: apiregistration.k8s.io/v1beta1
+  kind: APIService
+  metadata:
+    name: v1beta1.admission.online.openshift.io
+  spec:
+    caBundle: /etc/pki/CA/cacert.pem
+    group: admission.online.openshift.io
+    groupPriorityMinimum: 1000
+    versionPriority: 15
+    service:
+      name: server
+      namespace: <namespace_name>
+    version: v1beta1
+----
+
+. Define a webhook, which is in this example validating:
++
+[source,yaml]
+----
+- apiVersion: admissionregistration.k8s.io/v1beta1
+  kind: ValidatingWebhookConfiguration
+  metadata:
+    name: <resourcename>.admission.online.openshift.io
+  webhooks:
+  - name: <resourcename>.admission.online.openshift.io
+    clientConfig:
+      service:  <1>
+        namespace: default
+        name: kubernetes
+        path: /apis/admission.online.openshift.io/v1beta1/<resourcename>
+      caBundle: <CA_data_name>
+    rules:
+    - operations:
+      - CREATE
+      apiGroups:
+      - project.openshift.io
+      apiVersions:
+      - "*"
+      resources:
+      - projectrequests
+    - operations:
+      - CREATE
+      apiGroups:
+      - ""
+      apiVersions:
+      - "*"
+      resources:
+      - namespaces
+    failurePolicy: Fail
+----
+<1> Access the webhook through the aggregated API.
+
+. Deploy each of those YAML defined resources using `oc apply -f`:
```

Comment:
I have now included an `oc apply -f` command after each YAML example, along with filenames for each, so that they are applied in order. I will test this once my new Quicklab environment finishes building.

Reply:
I have successfully tested the whole procedure directly from the document in a new environment, now that we have split each element out into individual YAML files.